firebase-firestore/src/main/java/com/google/firebase/firestore/bundle/BundleSerializer.java

```diff
@@ -167,24 +167,35 @@ private int decodeLimit(JSONObject structuredQuery) {
     }
   }
 
-  private Bound decodeBound(@Nullable JSONObject bound) throws JSONException {
+  private Bound decodeStartAtBound(@Nullable JSONObject bound) throws JSONException {
```

Comment:
Hm. I wonder if we can offload the duplicated logic into decodePosition, or if that just gets us back to the same level of misdirection.

Reply:
We would need to somehow convey whether this is a `start` or `end` bound.